// microsoft-cognitiveservices-speech-sdk/distrib/lib/src/sdk/Exports.js
"use strict";
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT license.
Object.defineProperty(exports, "__esModule", { value: true });
var AudioConfig_1 = require("./Audio/AudioConfig");
exports.AudioConfig = AudioConfig_1.AudioConfig;
var AudioStreamFormat_1 = require("./Audio/AudioStreamFormat");
exports.AudioStreamFormat = AudioStreamFormat_1.AudioStreamFormat;
var AudioInputStream_1 = require("./Audio/AudioInputStream");
exports.AudioInputStream = AudioInputStream_1.AudioInputStream;
exports.PullAudioInputStream = AudioInputStream_1.PullAudioInputStream;
exports.PushAudioInputStream = AudioInputStream_1.PushAudioInputStream;
var AudioOutputStream_1 = require("./Audio/AudioOutputStream");
exports.AudioOutputStream = AudioOutputStream_1.AudioOutputStream;
exports.PullAudioOutputStream = AudioOutputStream_1.PullAudioOutputStream;
exports.PushAudioOutputStream = AudioOutputStream_1.PushAudioOutputStream;
var CancellationReason_1 = require("./CancellationReason");
exports.CancellationReason = CancellationReason_1.CancellationReason;
var PullAudioInputStreamCallback_1 = require("./Audio/PullAudioInputStreamCallback");
exports.PullAudioInputStreamCallback = PullAudioInputStreamCallback_1.PullAudioInputStreamCallback;
var PushAudioOutputStreamCallback_1 = require("./Audio/PushAudioOutputStreamCallback");
exports.PushAudioOutputStreamCallback = PushAudioOutputStreamCallback_1.PushAudioOutputStreamCallback;
var KeywordRecognitionModel_1 = require("./KeywordRecognitionModel");
exports.KeywordRecognitionModel = KeywordRecognitionModel_1.KeywordRecognitionModel;
var SessionEventArgs_1 = require("./SessionEventArgs");
exports.SessionEventArgs = SessionEventArgs_1.SessionEventArgs;
var RecognitionEventArgs_1 = require("./RecognitionEventArgs");
exports.RecognitionEventArgs = RecognitionEventArgs_1.RecognitionEventArgs;
var OutputFormat_1 = require("./OutputFormat");
exports.OutputFormat = OutputFormat_1.OutputFormat;
var IntentRecognitionEventArgs_1 = require("./IntentRecognitionEventArgs");
exports.IntentRecognitionEventArgs = IntentRecognitionEventArgs_1.IntentRecognitionEventArgs;
var RecognitionResult_1 = require("./RecognitionResult");
exports.RecognitionResult = RecognitionResult_1.RecognitionResult;
var SpeechRecognitionResult_1 = require("./SpeechRecognitionResult");
exports.SpeechRecognitionResult = SpeechRecognitionResult_1.SpeechRecognitionResult;
var IntentRecognitionResult_1 = require("./IntentRecognitionResult");
exports.IntentRecognitionResult = IntentRecognitionResult_1.IntentRecognitionResult;
var LanguageUnderstandingModel_1 = require("./LanguageUnderstandingModel");
exports.LanguageUnderstandingModel = LanguageUnderstandingModel_1.LanguageUnderstandingModel;
var SpeechRecognitionEventArgs_1 = require("./SpeechRecognitionEventArgs");
exports.SpeechRecognitionEventArgs = SpeechRecognitionEventArgs_1.SpeechRecognitionEventArgs;
exports.ConversationTranscriptionEventArgs = SpeechRecognitionEventArgs_1.ConversationTranscriptionEventArgs;
var SpeechRecognitionCanceledEventArgs_1 = require("./SpeechRecognitionCanceledEventArgs");
exports.SpeechRecognitionCanceledEventArgs = SpeechRecognitionCanceledEventArgs_1.SpeechRecognitionCanceledEventArgs;
var TranslationRecognitionEventArgs_1 = require("./TranslationRecognitionEventArgs");
exports.TranslationRecognitionEventArgs = TranslationRecognitionEventArgs_1.TranslationRecognitionEventArgs;
var TranslationSynthesisEventArgs_1 = require("./TranslationSynthesisEventArgs");
exports.TranslationSynthesisEventArgs = TranslationSynthesisEventArgs_1.TranslationSynthesisEventArgs;
var TranslationRecognitionResult_1 = require("./TranslationRecognitionResult");
exports.TranslationRecognitionResult = TranslationRecognitionResult_1.TranslationRecognitionResult;
var TranslationSynthesisResult_1 = require("./TranslationSynthesisResult");
exports.TranslationSynthesisResult = TranslationSynthesisResult_1.TranslationSynthesisResult;
var ResultReason_1 = require("./ResultReason");
exports.ResultReason = ResultReason_1.ResultReason;
var SpeechConfig_1 = require("./SpeechConfig");
exports.SpeechConfig = SpeechConfig_1.SpeechConfig;
exports.SpeechConfigImpl = SpeechConfig_1.SpeechConfigImpl;
var SpeechTranslationConfig_1 = require("./SpeechTranslationConfig");
exports.SpeechTranslationConfig = SpeechTranslationConfig_1.SpeechTranslationConfig;
exports.SpeechTranslationConfigImpl = SpeechTranslationConfig_1.SpeechTranslationConfigImpl;
var PropertyCollection_1 = require("./PropertyCollection");
exports.PropertyCollection = PropertyCollection_1.PropertyCollection;
var PropertyId_1 = require("./PropertyId");
exports.PropertyId = PropertyId_1.PropertyId;
var Recognizer_1 = require("./Recognizer");
exports.Recognizer = Recognizer_1.Recognizer;
var SpeechRecognizer_1 = require("./SpeechRecognizer");
exports.SpeechRecognizer = SpeechRecognizer_1.SpeechRecognizer;
var IntentRecognizer_1 = require("./IntentRecognizer");
exports.IntentRecognizer = IntentRecognizer_1.IntentRecognizer;
var VoiceProfileType_1 = require("./VoiceProfileType");
exports.VoiceProfileType = VoiceProfileType_1.VoiceProfileType;
var TranslationRecognizer_1 = require("./TranslationRecognizer");
exports.TranslationRecognizer = TranslationRecognizer_1.TranslationRecognizer;
var Translations_1 = require("./Translations");
exports.Translations = Translations_1.Translations;
var NoMatchReason_1 = require("./NoMatchReason");
exports.NoMatchReason = NoMatchReason_1.NoMatchReason;
var NoMatchDetails_1 = require("./NoMatchDetails");
exports.NoMatchDetails = NoMatchDetails_1.NoMatchDetails;
var TranslationRecognitionCanceledEventArgs_1 = require("./TranslationRecognitionCanceledEventArgs");
exports.TranslationRecognitionCanceledEventArgs = TranslationRecognitionCanceledEventArgs_1.TranslationRecognitionCanceledEventArgs;
var IntentRecognitionCanceledEventArgs_1 = require("./IntentRecognitionCanceledEventArgs");
exports.IntentRecognitionCanceledEventArgs = IntentRecognitionCanceledEventArgs_1.IntentRecognitionCanceledEventArgs;
var CancellationDetailsBase_1 = require("./CancellationDetailsBase");
exports.CancellationDetailsBase = CancellationDetailsBase_1.CancellationDetailsBase;
var CancellationDetails_1 = require("./CancellationDetails");
exports.CancellationDetails = CancellationDetails_1.CancellationDetails;
var CancellationErrorCodes_1 = require("./CancellationErrorCodes");
exports.CancellationErrorCode = CancellationErrorCodes_1.CancellationErrorCode;
var ConnectionEventArgs_1 = require("./ConnectionEventArgs");
exports.ConnectionEventArgs = ConnectionEventArgs_1.ConnectionEventArgs;
var ServiceEventArgs_1 = require("./ServiceEventArgs");
exports.ServiceEventArgs = ServiceEventArgs_1.ServiceEventArgs;
var Connection_1 = require("./Connection");
exports.Connection = Connection_1.Connection;
var PhraseListGrammar_1 = require("./PhraseListGrammar");
exports.PhraseListGrammar = PhraseListGrammar_1.PhraseListGrammar;
var DialogServiceConfig_1 = require("./DialogServiceConfig");
exports.DialogServiceConfig = DialogServiceConfig_1.DialogServiceConfig;
var BotFrameworkConfig_1 = require("./BotFrameworkConfig");
exports.BotFrameworkConfig = BotFrameworkConfig_1.BotFrameworkConfig;
var CustomCommandsConfig_1 = require("./CustomCommandsConfig");
exports.CustomCommandsConfig = CustomCommandsConfig_1.CustomCommandsConfig;
var DialogServiceConnector_1 = require("./DialogServiceConnector");
exports.DialogServiceConnector = DialogServiceConnector_1.DialogServiceConnector;
var ActivityReceivedEventArgs_1 = require("./ActivityReceivedEventArgs");
exports.ActivityReceivedEventArgs = ActivityReceivedEventArgs_1.ActivityReceivedEventArgs;
var ServicePropertyChannel_1 = require("./ServicePropertyChannel");
exports.ServicePropertyChannel = ServicePropertyChannel_1.ServicePropertyChannel;
var ProfanityOption_1 = require("./ProfanityOption");
exports.ProfanityOption = ProfanityOption_1.ProfanityOption;
var BaseAudioPlayer_1 = require("./Audio/BaseAudioPlayer");
exports.BaseAudioPlayer = BaseAudioPlayer_1.BaseAudioPlayer;
var ConnectionMessageEventArgs_1 = require("./ConnectionMessageEventArgs");
exports.ConnectionMessageEventArgs = ConnectionMessageEventArgs_1.ConnectionMessageEventArgs;
var ConnectionMessage_1 = require("./ConnectionMessage");
exports.ConnectionMessage = ConnectionMessage_1.ConnectionMessage;
var VoiceProfile_1 = require("./VoiceProfile");
exports.VoiceProfile = VoiceProfile_1.VoiceProfile;
var VoiceProfileEnrollmentResult_1 = require("./VoiceProfileEnrollmentResult");
exports.VoiceProfileEnrollmentResult = VoiceProfileEnrollmentResult_1.VoiceProfileEnrollmentResult;
exports.VoiceProfileEnrollmentCancellationDetails = VoiceProfileEnrollmentResult_1.VoiceProfileEnrollmentCancellationDetails;
var VoiceProfileResult_1 = require("./VoiceProfileResult");
exports.VoiceProfileResult = VoiceProfileResult_1.VoiceProfileResult;
exports.VoiceProfileCancellationDetails = VoiceProfileResult_1.VoiceProfileCancellationDetails;
var VoiceProfileClient_1 = require("./VoiceProfileClient");
exports.VoiceProfileClient = VoiceProfileClient_1.VoiceProfileClient;
var SpeakerRecognizer_1 = require("./SpeakerRecognizer");
exports.SpeakerRecognizer = SpeakerRecognizer_1.SpeakerRecognizer;
var SpeakerIdentificationModel_1 = require("./SpeakerIdentificationModel");
exports.SpeakerIdentificationModel = SpeakerIdentificationModel_1.SpeakerIdentificationModel;
var SpeakerVerificationModel_1 = require("./SpeakerVerificationModel");
exports.SpeakerVerificationModel = SpeakerVerificationModel_1.SpeakerVerificationModel;
var AutoDetectSourceLanguageConfig_1 = require("./AutoDetectSourceLanguageConfig");
exports.AutoDetectSourceLanguageConfig = AutoDetectSourceLanguageConfig_1.AutoDetectSourceLanguageConfig;
var AutoDetectSourceLanguageResult_1 = require("./AutoDetectSourceLanguageResult");
exports.AutoDetectSourceLanguageResult = AutoDetectSourceLanguageResult_1.AutoDetectSourceLanguageResult;
var SourceLanguageConfig_1 = require("./SourceLanguageConfig");
exports.SourceLanguageConfig = SourceLanguageConfig_1.SourceLanguageConfig;
var SpeakerRecognitionResult_1 = require("./SpeakerRecognitionResult");
exports.SpeakerRecognitionResult = SpeakerRecognitionResult_1.SpeakerRecognitionResult;
exports.SpeakerRecognitionResultType = SpeakerRecognitionResult_1.SpeakerRecognitionResultType;
exports.SpeakerRecognitionCancellationDetails = SpeakerRecognitionResult_1.SpeakerRecognitionCancellationDetails;
var Exports_1 = require("./Transcription/Exports");
exports.Conversation = Exports_1.Conversation;
exports.ConversationExpirationEventArgs = Exports_1.ConversationExpirationEventArgs;
exports.ConversationParticipantsChangedEventArgs = Exports_1.ConversationParticipantsChangedEventArgs;
exports.ConversationTranslationCanceledEventArgs = Exports_1.ConversationTranslationCanceledEventArgs;
exports.ConversationTranslationEventArgs = Exports_1.ConversationTranslationEventArgs;
exports.ConversationTranslationResult = Exports_1.ConversationTranslationResult;
exports.ConversationTranslator = Exports_1.ConversationTranslator;
exports.ConversationTranscriber = Exports_1.ConversationTranscriber;
exports.Participant = Exports_1.Participant;
exports.ParticipantChangedReason = Exports_1.ParticipantChangedReason;
exports.User = Exports_1.User;
var SpeechSynthesisOutputFormat_1 = require("./SpeechSynthesisOutputFormat");
exports.SpeechSynthesisOutputFormat = SpeechSynthesisOutputFormat_1.SpeechSynthesisOutputFormat;
var SpeechSynthesizer_1 = require("./SpeechSynthesizer");
exports.SpeechSynthesizer = SpeechSynthesizer_1.SpeechSynthesizer;
var SpeechSynthesisResult_1 = require("./SpeechSynthesisResult");
exports.SpeechSynthesisResult = SpeechSynthesisResult_1.SpeechSynthesisResult;
var SpeechSynthesisEventArgs_1 = require("./SpeechSynthesisEventArgs");
exports.SpeechSynthesisEventArgs = SpeechSynthesisEventArgs_1.SpeechSynthesisEventArgs;
var SpeechSynthesisWordBoundaryEventArgs_1 = require("./SpeechSynthesisWordBoundaryEventArgs");
exports.SpeechSynthesisWordBoundaryEventArgs = SpeechSynthesisWordBoundaryEventArgs_1.SpeechSynthesisWordBoundaryEventArgs;
var SpeakerAudioDestination_1 = require("./Audio/SpeakerAudioDestination");
exports.SpeakerAudioDestination = SpeakerAudioDestination_1.SpeakerAudioDestination;
var ConversationTranscriptionCanceledEventArgs_1 = require("./ConversationTranscriptionCanceledEventArgs");
exports.ConversationTranscriptionCanceledEventArgs = ConversationTranscriptionCanceledEventArgs_1.ConversationTranscriptionCanceledEventArgs;
