// microsoft-cognitiveservices-speech-sdk/distrib/es2015/src/common.speech/QueryParameterNames.js
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT license.
export class QueryParameterNames {
    static get TestHooksParamName() {
        return "testhooks";
    }
    static get ConnectionIdHeader() {
        return "X-ConnectionId";
    }
    static get DeploymentIdParamName() {
        return "cid";
    }
    static get FormatParamName() {
        return "format";
    }
    static get LanguageParamName() {
        return "language";
    }
    static get TranslationFromParamName() {
        return "from";
    }
    static get TranslationToParamName() {
        return "to";
    }
    static get Profanify() {
        return "profanity";
    }
    static get EnableAudioLogging() {
        return "storeAudio";
    }
    static get EnableWordLevelTimestamps() {
        return "wordLevelTimestamps";
    }
    static get InitialSilenceTimeoutMs() {
        return "initialSilenceTimeoutMs";
    }
    static get EndSilenceTimeoutMs() {
        return "endSilenceTimeoutMs";
    }
    static get StableIntermediateThreshold() {
        return "stableIntermediateThreshold";
    }
    static get StableTranslation() {
        return "stableTranslation";
    }
    static get EnableLanguageID() {
        return "lidEnabled";
    }
}
