// microsoft-cognitiveservices-speech-sdk/distrib/es2015/src/common/Emitter.js
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT license.
import { EventEmitter } from "ws";
// tslint:disable-next-line:max-classes-per-file
class CustomEmitter {
    constructor() {
        this.target = null;
        try {
            this.target = new EventTarget();
        }
        catch (error) {
            if (typeof window !== "undefined") {
                this.target = window.document.createElement("eventTarget");
            }
            else {
                throw error;
            }
        }
    }
    on(eventName, listener) {
        return this.target.addEventListener(eventName, listener);
    }
    off(eventName, listener) {
        return this.target.removeEventListener(eventName, listener);
    }
    emit(eventName, detail) {
        this.target.dispatchEvent(new CustomEvent(eventName, { detail }));
    }
}
// tslint:disable-next-line:max-classes-per-file
export class PlatformEmitter {
    constructor() {
        if (EventEmitter !== undefined) {
            this.privEmitter = new EventEmitter();
        }
        else {
            this.privEmitter = new CustomEmitter();
        }
    }
    on(eventName, fn) {
        this.privEmitter.on(eventName, fn);
    }
    off(eventName, fn) {
        this.privEmitter.off(eventName, fn);
    }
    emit(eventName, params) {
        this.privEmitter.emit(eventName, params);
    }
}
