// microsoft-cognitiveservices-speech-sdk/distrib/es2015/src/common/EventSource.js
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT license.
import { PlatformEmitter } from "./Emitter";
import { ObjectDisposedError } from "./Error";
import { createNoDashGuid } from "./Guid";
export class EventSource {
    constructor(metadata) {
        this.privEventIds = {};
        this.privIsDisposed = false;
        this.onEvent = (event) => {
            if (this.isDisposed()) {
                throw (new ObjectDisposedError("EventSource"));
            }
            if (this.metadata) {
                for (const paramName in this.metadata) {
                    if (paramName) {
                        if (event.metadata) {
                            if (!event.metadata[paramName]) {
                                event.metadata[paramName] = this.metadata[paramName];
                            }
                        }
                    }
                }
            }
            for (const eventId in this.privEventIds) {
                if (eventId && this.privEventIds[eventId]) {
                    this.privEmitter.emit(eventId, event);
                }
            }
        };
        this.attach = (onEventCallback) => {
            const id = createNoDashGuid();
            this.privEmitter.on(id, onEventCallback);
            this.privEventIds[id] = true;
            return {
                detach: () => {
                    this.privEmitter.off(id, onEventCallback);
                    delete this.privEventIds[id];
                    return Promise.resolve();
                },
            };
        };
        this.attachListener = (listener) => {
            return this.attach(listener.onEvent);
        };
        this.isDisposed = () => {
            return this.privIsDisposed;
        };
        this.dispose = () => {
            this.privIsDisposed = true;
        };
        this.privMetadata = metadata;
        this.privEmitter = new PlatformEmitter();
    }
    get metadata() {
        return this.privMetadata;
    }
}
